### PR TITLE
[Rebase & FF] Integrate MM_SUPV into Q35 platform

### PR DESCRIPTION
**Pull request by kuqin12**
This change updated original OVMF package code to extend the support for MM_STANDALONE drivers for x86 systems. The dynamic PCDs are replaced by check-and-assert logic.

The system is verified bootable to Windows on Q35 system.

### CODE REVIEW

File: Platforms/QemuQ35Pkg/PlatformPei/Platform.c [hunk 1]
```diff
@@ -433,18 +450,22 @@ MiscInitialization (
...
-                    PcdOvmfHostBridgePciDevId,
-                    MICROVM_PSEUDO_DEVICE_ID
-                    );
+      PcdStatus = EFI_UNSUPPORTED;
+      // MU_CHANGE: Remove dynamic PCD set to support usage in Standalone MM
+      // PcdStatus = PcdSet16S (
+      //               PcdOvmfHostBridgePciDevId,
+      //               MICROVM_PSEUDO_DEVICE_ID
+      //               );
       ASSERT_RETURN_ERROR (PcdStatus);
```
Comment (os-d) [inline Platforms/QemuQ35Pkg/PlatformPei/Platform.c:459]: Maybe we never hit these cases, but won't this and the next one assert 100% of the time?

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c [hunk 2]
```diff
@@ -1129,7 +1131,10 @@ FvbInitialize (
   //
   InstallVirtualAddressChangeHandler ();
 
-  PcdStatus = PcdSetBoolS (PcdOvmfFlashVariablesEnable, TRUE);
-  ASSERT_RETURN_ERROR (PcdStatus);
+  // MU_CHANGE: Abstract dynamic PCD set to support Standalone MM
+  // PcdStatus = PcdSetBoolS (PcdOvmfFlashVariablesEnable, TRUE);
+  // ASSERT_RETURN_ERROR (PcdStatus);
```
Comment (os-d) [inline Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c:1136]: For all the commented out code segments can we just remove that code? Or do you think there is value in keeping the old code as reference for what was changed?

File: Platforms/QemuQ35Pkg/PlatformPei/Platform.c (hunk 1 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/PlatformPei/Platform.c:459]: Yes :) It is basically saying that we are not supporting types other than Q35.

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c (hunk 2 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c:1136]: I think it makes sense now. This was made before the removal of OvmfPkg, I will go through them and see if what can be cleaned up in that regard.

**Review by apop5 — APPROVED** [review]

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h [hunk 3]
```diff
@@ -194,4 +194,9 @@ SetPcdFlashNvStorageBaseAddresses (
   VOID
   );
 
+VOID
+UpdateOvmfFlashVariablesEnable (
```
Comment (kenlautner) [inline Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h:198]: Any particular reason why you have Ovmf in the name of this new function?  Given that we've gotten rid of the package.

File: Platforms/QemuQ35Pkg/QemuQ35Pkg.ci.yaml [hunk 4]
```diff
@@ -42,7 +43,7 @@
             "SetupDataPkg/SetupDataPkg.dec",
             "MfciPkg/MfciPkg.dec",
             "OemPkg/OemPkg.dec",
-            "QemuQ35Pkg/QemuQ35Pkg.dec"
+            "QemuQ35Pkg/QemuQ35Pkg.dec",
```
Comment (kenlautner) [inline Platforms/QemuQ35Pkg/QemuQ35Pkg.ci.yaml:46]: This comma seems unnecessary.

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h (hunk 3 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h:198]: Not anymore~ will update.

**Review by kenlautner — APPROVED** [review]

File: Platforms/QemuQ35Pkg/QemuQ35Pkg.ci.yaml (hunk 4 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/QemuQ35Pkg.ci.yaml]: Removed.

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h (hunk 3 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.h]: Renamed to QEMU flash variable.

File: Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c (hunk 2 above)
Comment (kuqin12) [reply Platforms/QemuQ35Pkg/QemuFlashFvbServicesRuntimeDxe/FwBlockService.c]: Went through all commits another round. The comments should be removed now.

**Pull request merged**